### ISSUE TITLE
Trying to open with Visual Studio

### ISSUE THREAD

**Issue by Moonslate** [visualboyadvance-m/visualboyadvance-m#592]
I'm trying to open the project for almost one week, but I'm having very troubles with it. Can you please generate a Visual Studio solution and send me?

**Comment by ZachBacon** [comment]
Open in visual studio the cmakelists.txt and visual studio will run the necessary commands to cmske (assuming you installed the cmake module for visual studio.)

**Comment by Moonslate** [comment]
> Open in visual studio the cmakelists.txt and visual studio will run the necessary commands to cmske (assuming you installed the cmake module for visual studio.)

Is that what I was trying, but always fails in dependences. And it is downloading and building several dependences that I already have into my computer. Can you send the solution please?

**Comment by ZachBacon** [comment]
The solution is generated by cmake, we make use of vcpkg as we need to make sure the build evnvironment is what we need it to be to build vba-m so as for a pre made solution file, you'll have none, it's all setup by cmake itself.

**Comment by Moonslate** [comment]
Yes, I'm talking about the solution generated by CMake when you build, with this solution I can link manually the libraries that are needed.

**Comment by ZachBacon** [comment]
@rkitover mind taking over?

**Comment by denisfa** [comment]
@Moonslate I just did a clean build today using VS2019. Please do try again, using latest commit. Check [here](https://github.com/visualboyadvance-m/visualboyadvance-m/blob/dev-manual/DEVELOPER-MANUAL.md) for a WIP about using visual studio for development. It will use cmake, at least for now.

**Comment by Moonslate** [comment]
I tried, and the same error occurred with vcpkg. But this time I killed the Visual Studio's process, and after open again, the vcpkg started working. But now I want to disable the automatic libraries downloading. How I do that? There is 9 dependence in the list, but the vcpkg download and build 19. My computer is a 10 years old notebook, so it is very slow, and have low storage. I already have all libraries required by VBA-M, and I want to use those instead of having to build every single lib in different configurations again. Because of that I was asking for the solution, that can be generated with CMake, for linking manually. I started the downloading, and after several minutes, cancelled because was taking so much time, and the vcpkg folder was with 2GB! 2GB for one repository (VBA-M) only.

**Comment by rkitover** [comment]
@Moonslate just edit `CMakeSettings.json` and remove the `VCPKG_TARGET_TRIPLET` and pass the necessary cmake variables pointing to your libraries.

We cannot support user-built libraries for Visual Studio because there is no standard way to do that.

You may have to dig into the cmake modules and google around to find the right cmake variables to set, but it should be doable.

If you get this to work, I'd appreciate it if you paste your whole cmake command here and we will document it for others in this situation.

**Comment by rkitover** [comment]
Also I should note that the dependencies are only built once, and when there are updates. Once vcpkg builds all the dependencies they will be reused on subsequent builds.

**Comment by rkitover** [comment]
@Moonslate if you want to play with cmake from the commandline instead, I am working on developer manual in another branch which has a guide for setting up powershell:

https://github.com/visualboyadvance-m/visualboyadvance-m/blob/dev-manual/DEVELOPER-MANUAL.md

**Comment by Moonslate** [comment]
> 
> 
> @Moonslate just edit `CMakeSettings.json` and remove the `VCPKG_TARGET_TRIPLET` and pass the necessary cmake variables pointing to your libraries.
> 

@rkitover, this worked.
Before:
`"cmakeCommandArgs": "-DVCPKG_TARGET_TRIPLET=x86-windows -DENABLE_SDL=TRUE"`
And after:
`"cmakeCommandArgs": -DENABLE_SDL=TRUE"`

Now I can link the libraries manually in the `CMake Setting` as you can see in my screenshot:
![screenshot](https://user-images.githubusercontent.com/42197658/72684549-4856c780-3ac0-11ea-84d2-fddee0bb1417.png)

> Also I should note that the dependencies are only built once, and when there are updates. Once 
> vcpkg builds all the dependencies they will be reused on subsequent builds.

I know, but this still not a advantage, since I already have all of the dependencies. Thanks for helping.
Now I'll try to build after liking all libraries, but I have bad luck, so I guess I will come back here.

**Comment by Moonslate** [comment]
Ok, I was linking the libraries, but I still need to link the `vcpkgconfig` executable:
`1> [CMake] -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) `

And it still wanting to execute vcpkg, but I removed the `-DVCPKG_TARGET_TRIPLET=x86-windows`.

**Comment by rkitover** [comment]
pkg-config is a different thing than vcpkg and that is not a fatal error. pkg-config is a system for getting flags for libraries on unix systems and mingw.

Not finding pkg-config is not a fatal error.

You should be able to build the project using your own libraries, if you have any issues I will try to help.

**Comment by Moonslate** [comment]
> Not finding pkg-config is not a fatal error.
> You should be able to build the project using your own libraries,

**Comment by ZachBacon** [comment]
@Moonslate I'm going to leave this issue open, if you could provide feed back to help improve the build experience for those in your situation where you have the libraries already, it would be a great help.

**Comment by Moonslate** [comment]
Sorry, I closed the issue accidentally while typing.

@rkitover, even if this is not a fatal error, the Visual Studio don't compile. Maybe I'm missing a error-level configuration?

**Comment by rkitover** [comment]
@Moonslate please provide full logs, you can post them on gist for example.

**Comment by rkitover** [comment]
@Moonslate and your `CMakeCache.txt` too if possible, that would help a lot.

**Comment by Moonslate** [comment]
Hello, sorry for taking so long to reply. I give up now. I tried for several days and I'm tired. I never used `CMake` before, and your `CMakeLists.txt` has so many things together that is so hard to configure. There is the `vcpkg` that download a huge amount of GB of unnecessary (since I already have almost all) libs, there is error with git in the `CMakeList.txt` (Why git is needed to compile?) There is libs that even disabling the feature is still needed and I can't generate the `CMakeCache.txt` without the libs linked.

I deleted from line 205 to line 207 of `CMakeLists.txt` to solve a problem with git. 
> 
> 
> @Moonslate and your `CMakeCache.txt` too if possible, that would help a lot.

I opened this issue because I was having problem to generate the cache, so there is no `CMakeCache.txt`.

**Comment by ZachBacon** [comment]
Git is needed as there's a submodule containing some needed stuff to compile, even on the msvc side of things, it's mainly headers etc.

**Comment by rkitover** [comment]
You just need to install git for windows or the git package from chocolatey, just about every windows developer has this so I don't consider this an issue. There is even an option nowadays in visual studio to install git along with everything else.

However, if you want to build from a source that is not a git clone, make sure to put the dependencies submodule in the source directory.

> I opened this issue because I was having problem to generate the cache, so there is no CMakeCache.txt.

The file is always generated, even on failed runs, it is in the cmake build directory.

You might also try building with msys2, that may be easier for you. There are even ways to configure visual studio to use msys2 for a project.

**Comment by Moonslate** [comment]
@rkitover, I have git. But I was having a problem, and I had to delete that lines.

> The file is always generated, even on failed runs, it is in the cmake build directory.

I think I checked this folder and there was not CMakeCache.txt, but I'll check again later.

**Comment by Moonslate** [comment]
@rkitover, I have git. But I was having a problem, and I had to delete that lines.

> The file is always generated, even on failed runs, it is in the cmake build directory.

I think I checked this folder and there was not CMakeCache.txt, but I'll check again later.

**Comment by rkitover** [comment]
If the file is not there, are other log files that can give you some clues:

```
CMakeFiles/CMakeError.log
CMakeFiles/CMakeOutput.log
```
